docs/core-concepts/saturation-penalty.md

```diff
@@ -0,0 +1,80 @@
+---
+sidebar_position: 10
+---
+
+# Over-Saturation Penalty
+
+The penalty system in the saturation library is designed to incentivize proper risk management. It not only aims to fairly compensate LPs during various states of borrow and saturation utilization, but also prevents cascading liquidations. Therefore, a dynamic risk pricing system is required to create a self-balancing mechanism that maintains the protocol health.
+
+## Time-based penalty accrual
+
+The protocal calculates penalty accrual using a time based system. `accruePenalties` function in Saturation.sol takes a `duration` parameter that is defined as the `delta` between the current time and the last time penalty was accrued. See the call stack in `AmmalgamPair.sol`: `accrueSaturationPenaltiesAndInterest -> mintPenalties -> accruePenalties`.
```

Comment:
Can we hyperlink to the contract docs for the mentioned methods?
For example: [`accruePenalties`](https://docs.ammalgam.xyz/docs/developer-guide/contracts/libraries/Saturation.sol/library.Saturation#accruepenalties)